MSAL/test/unit/MSALTokenCacheAccessorTests.m

```diff
@@ -121,284 +136,260 @@ - (void)tearDown {
-    XCTAssertEqualObjects(atItem2.user.userIdentifier, atItemInCache2.user.userIdentifier);
-    XCTAssertEqualObjects(atItem2.tenantId, atItemInCache2.tenantId);
-    XCTAssertTrue(atItem2.isExpired==atItemInCache2.isExpired);
-    XCTAssertEqualObjects([atItem2 tokenCacheKey:nil].service, [atItemInCache2 tokenCacheKey:nil].service);
-    XCTAssertEqualObjects([atItem2 tokenCacheKey:nil].account, [atItemInCache2 tokenCacheKey:nil].account);
-    XCTAssertEqualObjects(atItem2.clientId, atItemInCache2.clientId);
-    XCTAssertEqualObjects(atItem2.clientInfo.uid, atItemInCache2.clientInfo.uid);
-    XCTAssertEqualObjects(atItem2.clientInfo.utid, atItemInCache2.clientInfo.utid);
+    //there should be still one AT and one RT in cache
+    NSArray <MSALAccessTokenCacheItem *> *atsInCache = [_cache.dataSource getAccessTokenItemsWithKey:nil context:nil error:nil];
+    XCTAssertEqual(atsInCache.count, 1);
+    NSArray <MSALRefreshTokenCacheItem *> *rtsInCache = [_cache.dataSource allRefreshTokens:nil context:nil error:nil];
+    XCTAssertEqual(rtsInCache.count, 1);
 }
 
-- (void)testSaveAndRetrieveRefreshToken {
+- (void)testSaveMultipleTokens
+{
+    //save first and second AT/RT
+    NSError *error;
+    [_cache saveAccessAndRefreshToken:_requestParam1 response:_testTokenResponse context:nil error:&error];
+    XCTAssertNil(error);
     
-    //prepare request parameters
-    MSALRequestParameters *requestParam = [MSALRequestParameters new];
-    requestParam.unvalidatedAuthority = _testAuthority;
-    requestParam.clientId = _testClientId;
-    [requestParam setScopesFromArray:@[@"User.Read"]];
-    requestParam.user = _user1;
+    error = nil;
+    [_cache saveAccessAndRefreshToken:_requestParam2 response:_testTokenResponse2 context:nil error:&error];
+    XCTAssertNil(error);
     
-    //prepare token response and save AT/RT
-    MSALRefreshTokenCacheItem *rtItem = [[MSALRefreshTokenCacheItem alloc] initWithEnvironment:_testAuthority.msalHostWithPort
-                                                                                      clientId:_testClientId
-                                                                                      response:_testTokenResponse];
-    [_cache saveAccessAndRefreshToken:requestParam response:_testTokenResponse context:nil error:nil];
+    //there should be two ATs and RTs in cache
+    NSArray <MSALAccessTokenCacheItem *> *atsInCache = [_cache.dataSource getAccessTokenItemsWithKey:nil context:nil error:nil];
+    NSArray <MSALRefreshTokenCacheItem *> *rtsInCache = [_cache.dataSource allRefreshTokens:nil context:nil error:nil];
+    XCTAssertEqual(atsInCache.count, 2);
+    XCTAssertEqual(rtsInCache.count, 2);
     
-    //retrieve RT
-    MSALRefreshTokenCacheItem *rtItemInCache = [_cache findRefreshToken:requestParam context:nil error:nil];
+    atsInCache = [atsInCache sortedArrayUsingComparator:^NSComparisonResult(id a, id b) {
```

Comment:
You don't have to do this, but it's good to know. If you implement both `-isEqual:` and `-hash` on a class you can dump them into NSSet and is isEqual on the sets to see if the two are the same. :)